### PR TITLE
[WIP] ONe Public IP test refactoring: get, delete

### PR DESCRIPTION
**Pull request by pauloewerton**
this change adds the ONe public ip plugin unit tests related to the get and delete methods. to test: check out code; run plugin tests.

### CODE REVIEW

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java [hunk 1]
```diff
@@ -139,296 +128,463 @@ public void testHasFailedUnsuccessful() {
...
+		Mockito.verify(this.response, Mockito.times(TestUtils.RUN_ONCE)).isError();
+	}
 
-		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, VerificationModeFactory.times(1));
-		OpenNebulaClientUtil.getVirtualMachine(Mockito.eq(client), Mockito.eq(virtualMachineId));
+	// test case: when invoking deleteInstance with valid order and cloud user, the plugin
+	// should retrieve the respective ONe virtual machine, shut it off, detach its public ip nic,
+	// delete the fogbow security group and finally delete the public ip instance network reserve
+	@Test
+	public void testDeleteInstance() throws FogbowException {
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java]: This test is failing...
Please check and correct.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java [hunk 2]
```diff
@@ -139,296 +128,463 @@ public void testHasFailedUnsuccessful() {
...
-	
-	// test case: When calling the deleteSecurityGroup method and not get a
-	// satisfactory response from the cloud, the resource will not be removed.
+
+	// test case: when invoking doDeleteInstance the plugin should retrieve the respective ONe
+	// virtual network and delete it
 	@Test
-	public void testDeleteSecurityGroupUnsuccessfully() throws UnexpectedException, UnauthorizedRequestException,
-			InvalidParameterException, InstanceNotFoundException {
+	public void testDoDeleteInstance() throws UnexpectedException, InstanceNotFoundException, InvalidParameterException,
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java]: This test has a runtime error...
Please check and correct.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java [hunk 3]
```diff
@@ -139,296 +128,463 @@ public void testHasFailedUnsuccessful() {
...
+		OpenNebulaClientUtil.getVirtualNetwork(Mockito.eq(this.client), Mockito.eq(this.instanceId));
+
+		Mockito.verify(this.virtualNetwork, Mockito.times(TestUtils.RUN_ONCE)).delete();
+		Mockito.verify(this.response, Mockito.times(TestUtils.RUN_ONCE)).isError();
+	}
+
+	// test case: when invoking doDeleteInstance with invalid parameters the plugin should
+	// throw an UnexpectedException
+	@Test
+	public void testDoDeleteInstanceFail() throws InstanceNotFoundException, InvalidParameterException,
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java]: This test has a runtime error...
Please check and correct.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java [hunk 4]
```diff
@@ -139,296 +128,463 @@ public void testHasFailedUnsuccessful() {
...
-	
-	// test case: When calling the detachNetworkInterfaceConnected method and cannot
-	// detach the NIC of the virtual machine passed by parameter, this will fail.
+
+	// test case: when invoking getInstance with valid public ip order and cloud user, the plugin
+	// should instantiate the respective fogbow public ip instance and return it
 	@Test
-	public void testDetachNetworkInterfaceConnectedUnsuccessfully()
-			throws InvalidParameterException, UnauthorizedRequestException, InstanceNotFoundException {
+	public void testGetInstance() throws FogbowException {
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java]: This test has a runtime error...
Please check and correct.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java (hunk 1 above)
Comment (pauloewerton) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java]: done

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java (hunk 3 above)
Comment (pauloewerton) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java]: done

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java (hunk 4 above)
Comment (pauloewerton) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java]: done

**Comment by pauloewerton** [comment]
@wassoares thanks for your review. comments addressed.

**Pull request merged**